### PR TITLE
OpenMPTarget: Adding a workaound for team scan.

### PR DESCRIPTION
**Pull request by rgayatri23**
The PR avoids the non-working team_scan code path.
The strategy it uses is to make one thread per team do the work.

### CODE REVIEW

File: core/src/OpenMPTarget/Kokkos_OpenMPTarget_Exec.hpp [hunk 1]
```diff
@@ -1584,22 +1591,36 @@ KOKKOS_INLINE_FUNCTION void parallel_scan(
...
+    for (iType i = start; i < end; ++i) {
+      lambda(i, scan_val, true);
+    }
+  }
+#pragma omp barrier
+#else
+  const auto nchunk = (end - start + team_size - 1) / team_size;
+  value_type accum  = 0;
+  each team has to process one or
+      more chunks of the prefix scan for (iType i = 0; i < nchunk; ++i) {
```
Comment (Rombur) [inline core/src/OpenMPTarget/Kokkos_OpenMPTarget_Exec.hpp:1623]: You removed the `//` and now the indentation is off. When we will remove the new code, the old code won't compile.

File: core/src/OpenMPTarget/Kokkos_OpenMPTarget_Exec.hpp [hunk 2]
```diff
@@ -1616,6 +1637,12 @@ KOKKOS_INLINE_FUNCTION void parallel_scan(
     // broadcast last value to rest of the team
     member.team_broadcast(accum, team_size - 1);
   }
+#endif
+
+  // for (iType i = 0; i < nchunk; ++i) {
+  // if(omp_get_thread_num() == 0)
+  // lambda(i, accum, true);
+  //}
```
Comment (Rombur) [inline core/src/OpenMPTarget/Kokkos_OpenMPTarget_Exec.hpp:1645]: Does this need to be removed or in the `#else`?

File: core/src/OpenMPTarget/Kokkos_OpenMPTarget_Exec.hpp (hunk 1 above)
Comment (rgayatri23) [reply core/src/OpenMPTarget/Kokkos_OpenMPTarget_Exec.hpp]: Thanks. Fixed it. Did not notice that the comments got uncommented in the blocked code path.

File: core/src/OpenMPTarget/Kokkos_OpenMPTarget_Exec.hpp (hunk 2 above)
Comment (rgayatri23) [reply core/src/OpenMPTarget/Kokkos_OpenMPTarget_Exec.hpp]: Thats needs to be removed. Done.

**Review by Rombur — APPROVED** [review]

**Review by crtrott — APPROVED** [review]

**Pull request merged**